Add support for defining default values as option for ActiveSupport::Configurable accessors

diegotoral opened the pull request:
### Summary
Sometimes it can be very strange or long have to define default values for config accessors as blocks, specially for simple values. This commit adds the ability to specify those values by just passing a ` default` option when defining the accessor.

p8 commented:
Hi @diegotoral thanks for the PR.
Can you give some code examples that would benefit from this change?

diegotoral commented:
Hi @p8!

The idea is to make simpler to specify default values for config accessors. The current implementation requires the usage of a block for specifying those values which results in a many lines of code.

```ruby
class User
  include ActiveSupport::Configurable

  config_accessor :a, default: 1
  config_accessor :b, default: 2
  config_accessor :c, default: 3

  # vs
  
  config_accessor :a do
    1
  end

  config_accessor :b do
    2
  end

  config_accessor :c do
    3
  end
end
```

jonathanhefner commented on activesupport/lib/active_support/configurable.rb:
```diff
@@ -118,7 +120,12 @@ def config_accessor(*names, instance_reader: true, instance_writer: true, instan
             class_eval writer, __FILE__, writer_line if instance_writer
           end
-          send("#{name}=", yield) if block_given?
+
+          if block_given?
+            send("#{name}=", yield)
+          else
+            send("#{name}=", default)
+          end
```
I don't have a strong preference, but what do you think of the more concise:

```suggestion
          send("#{name}=", block_given? ? yield : default)
```

diegotoral replied:
I like it! I will apply this change

jonathanhefner reviewed (commented):
I think this makes sense for parity with [`class_attribute`](https://api.rubyonrails.org/v6.1/classes/Class.html#method-i-class_attribute), which also has the `instance_reader`, `instance_writer`, and `instance_accessor` options.

diegotoral commented:
@p8 @jonathanhefner updated the branch to apply the requested changes. :smile:

diegotoral committed:
ActiveSupport::Configurable default value option

Sometimes it can be very strange or long have to define default values
for config accessors as blocks, specially when config is a simple value
like 1, true, :symbol. This commit adds the ability to specify those
values by just passing a ` default` option when defining the accessor.

It also makes config accessor's interface similar to other Rails
methods like `class_attribute`, which also has the instance_reader,
instance_writer and instance_accessor options.
